src/main/java/com/eerussianguy/firmalife/registry/VanillaStemStateMapper.java

```diff
@@ -0,0 +1,65 @@
+package com.eerussianguy.firmalife.registry;
+
+import com.eerussianguy.firmalife.blocks.BlockStemCrop;
+import com.google.common.collect.Maps;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockStem;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.properties.PropertyInteger;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.block.statemap.StateMapperBase;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+
+import java.util.Map;
+
+import static net.dries007.tfc.objects.blocks.agriculture.BlockCropTFC.WILD;
+
+public class VanillaStemStateMapper extends StateMapperBase
+{
+
+    //tfc data says facing=[dir] and stage=[0..maxStage]
+    //vanilla can render a vertical stalk wih facing=up and age=[0..7]
+    //or a bent stalk with facing=[dir] and no other parameters.
+    //visually, we want stage=maxStage to render a bent stalk
+    //and we want stage=0 to render age=0, and we want stage=maxStage-1 to render age=7
+    //for all other stages we can interpolate
+    //this returns the vanilla age from 0 to 7, or -1 if fully grown
+    public static int getVanillaAge(IBlockState state)
+    {
+        BlockStemCrop block = (BlockStemCrop)state.getBlock();
+        PropertyInteger stage = block.getStageProperty();
+        int growthStage = state.getValue(stage);
+        int maxStage = block.getCrop().getMaxStage();
+        if(growthStage == maxStage)
+            return -1;
+        float fractionGrown = ((float) growthStage) / ((float) (maxStage - 1));
+        return (int)(fractionGrown * 7.0f);
+    }
+    @Override
+    public ModelResourceLocation getModelResourceLocation(IBlockState state) {
+
+        BlockStemCrop block = (BlockStemCrop)state.getBlock();
+
+        //use the vanilla melon stem blockstates json
+        String s = Block.REGISTRY.getNameForObject(Blocks.MELON_STEM).toString();
+        Map<IProperty<?>, Comparable<? >> map = Maps. < IProperty<?>, Comparable<? >> newLinkedHashMap(state.getProperties());
```

Comment:
I am assuming this whole class is a workaround re: using vanilla's blockstates and coloring system? Since it works, I don't particularly care, but it seems like a lot for not a ton of gain

Reply:
It's a way to get the vanilla behavior without building anything into BlockStemCrop